minio/parsers.py

```diff
@@ -141,6 +141,39 @@ def parse_list_buckets(data):
         raise InvalidXMLError('Invalid XML provided for "ListBucketsResult" '
                               'some fields are missing. Message: {0}'.format(error.message))
 
+def _parse_objects_from_xml_elts(bucket_name, contents, common_prefixes):
+    """Internal function that extracts objects and common prefixes from
+    list_objects responses.
+    """
+    objects = []
+    for content in contents:
+        key_elt = content.find('s3:Key', _S3_NS)
+        object_name = urldecode(key_elt.text) if key_elt != None else None
```

Comment:
You could tell me why we need to do take care of `None`? `urldecode(key_elt.text)` would throw `NoneType` exception which means the xml format is invalid.